### PR TITLE
[frontend] Create analyst workbench from an entity (#3188)

### LINKED ISSUE #3188: Be able to create an analyst workbench from an entity
## Use case

In the "Data" section of an entity, the top right panel is for attached pending STIX bundles (aka Analyst workbenches).

image

There is no reason to not having a "+" button here to be able to create an empty workbench (aka empty STIX JSON file in pending mode) from this screen.

### PR DESCRIPTION
**Pull request by helene-nguyen**
### Proposed changes

* Extract WorkbenchFileCreator component to create a new analyst workbench
* Add entity property to be able to create a workbench from an entity : create the workbench file with stix entity in objects data and entityId in the file metadata.
* Bug fix : workbench created from a container entity should link all added entities to the container (in object_refs)

### Related issues

* https://github.com/OpenCTI-Platform/opencti/issues/3188

### Checklist

- [x] I consider the submitted work as finished
- [x] I tested the code for its functionality
- [ ] I wrote test cases for the relevant uses case
- [ ] I added/update the relevant documentation (either on github or on notion)
- [x] Where necessary I refactored code to improve the overall quality

### CODE REVIEW

File: opencti-platform/opencti-front/src/private/components/common/files/workbench/WorkbenchFileViewer.jsx
Comment (Kedae) [inline opencti-platform/opencti-front/src/private/components/common/files/workbench/WorkbenchFileViewer.jsx]: I think this is a miss from a LONG time ago. We have many more container than just reports, can you check that ?

File: opencti-platform/opencti-front/src/private/components/common/files/workbench/WorkbenchFileCreator.tsx
Comment (Kedae) [inline opencti-platform/opencti-front/src/private/components/common/files/workbench/WorkbenchFileCreator.tsx]: onCompleted?.();

File: opencti-platform/opencti-front/src/private/components/common/files/workbench/WorkbenchFileContent.jsx
Comment (Kedae) [inline opencti-platform/opencti-front/src/private/components/common/files/workbench/WorkbenchFileContent.jsx]: Could be simplified : currentEntityContainer.object_refs = R.uniq([...(currentEntityContainer.object_refs ?? []), ...objectIds ]);

File: opencti-platform/opencti-front/src/private/components/common/files/workbench/WorkbenchFileContent.jsx
Comment (Kedae) [inline opencti-platform/opencti-front/src/private/components/common/files/workbench/WorkbenchFileContent.jsx]: Can be simplified : const objectIds = [...stixDomainObjects, ...stixCyberObservables].map(({ id }) => id);

File: opencti-platform/opencti-front/src/private/components/common/files/workbench/WorkbenchFileContent.jsx
Comment (SouadHadjiat) [reply opencti-platform/opencti-front/src/private/components/common/files/workbench/WorkbenchFileContent.jsx]: `currentEntityContainer.object_refs ?? []` doesn't work because I had cases where currentEntityContainer.object_refs was an empty object {}, not an empty array. So I had to test that the object has a length to be sure that it's an array.

Comment (Kedae) [reply opencti-platform/opencti-front/src/private/components/common/files/workbench/WorkbenchFileContent.jsx]: That's weird... In that case use Array.isArray(currentEntityContainer). It is better because it tells what we want to test (if it actually is an array)

File: opencti-platform/opencti-front/src/private/components/common/files/workbench/WorkbenchFileCreator.tsx
Comment (SouadHadjiat) [reply opencti-platform/opencti-front/src/private/components/common/files/workbench/WorkbenchFileCreator.tsx]: hum, I don't see this very often in the code, I'm not sure everyone will understand this syntax, the if is more understandable

Comment (Kedae) [reply opencti-platform/opencti-front/src/private/components/common/files/workbench/WorkbenchFileCreator.tsx]: We don't see it often because it is rather new, but we want to use it, the same way as we use ?? or object?.property

File: opencti-platform/opencti-front/src/private/components/import/ImportContent.jsx
Comment (Kedae) [inline opencti-platform/opencti-front/src/private/components/import/ImportContent.jsx]: Why is !! needed ? Isn't that a boolean ?

**Comment by Kedae** [comment]
image
Could you align items on the line ?

File: opencti-platform/opencti-front/src/private/components/common/files/workbench/WorkbenchFileCreator.tsx
Comment (helene-nguyen) [reply opencti-platform/opencti-front/src/private/components/common/files/workbench/WorkbenchFileCreator.tsx]: Done

File: opencti-platform/opencti-front/src/private/components/common/files/workbench/WorkbenchFileViewer.jsx
Comment (SouadHadjiat) [reply opencti-platform/opencti-front/src/private/components/common/files/workbench/WorkbenchFileViewer.jsx]: isContainer is only used to set `disableImport` of WorkbenchFileLine, but `disableImport` doesn't seem to be used at all in WorkbenchFileLine... do you know why ? @Kedae I guess we should just remove it ?

Comment (Kedae) [reply opencti-platform/opencti-front/src/private/components/common/files/workbench/WorkbenchFileViewer.jsx]: Yep, didn't see that. You can simply remove it

File: opencti-platform/opencti-front/src/private/components/import/ImportContent.jsx
Comment (helene-nguyen) [reply opencti-platform/opencti-front/src/private/components/import/ImportContent.jsx]: displayCreate is not typed as boolean

image

Comment (Kedae) [reply opencti-platform/opencti-front/src/private/components/import/ImportContent.jsx]: That must be simply IntelliJ yelling because you are in a javascript and not typescript file.

File: opencti-platform/opencti-front/src/private/components/common/files/workbench/WorkbenchFileContent.jsx
Comment (helene-nguyen) [reply opencti-platform/opencti-front/src/private/components/common/files/workbench/WorkbenchFileContent.jsx]: Done

**Comment by SouadHadjiat** [comment]
> image Could you align items on the line ?

@Kedae not so simple to align, because the component is shared (used in /dashboard/import and it's aligned there)

Capture d'écran 2023-06-27 100545

File: opencti-platform/opencti-front/src/private/components/common/files/workbench/WorkbenchFileViewer.jsx
Comment (helene-nguyen) [reply opencti-platform/opencti-front/src/private/components/common/files/workbench/WorkbenchFileViewer.jsx]: Done

File: opencti-platform/opencti-front/src/private/components/common/files/workbench/WorkbenchFileContent.jsx
Comment (helene-nguyen) [reply opencti-platform/opencti-front/src/private/components/common/files/workbench/WorkbenchFileContent.jsx]: Done

File: opencti-platform/opencti-front/src/private/components/import/ImportContent.jsx
Comment (helene-nguyen) [reply opencti-platform/opencti-front/src/private/components/import/ImportContent.jsx]: Done

**Comment by SouadHadjiat** [comment]
PR status : Ready for more reviews and waiting for approval.

File: opencti-platform/opencti-front/src/private/components/common/files/workbench/WorkbenchFileContent.jsx
Comment (SarahBocognano) [inline opencti-platform/opencti-front/src/private/components/common/files/workbench/WorkbenchFileContent.jsx:375]: Can we get rid of Ramda ?

File: opencti-platform/opencti-front/src/private/components/common/files/workbench/WorkbenchFileLine.jsx
Comment (SarahBocognano) [inline opencti-platform/opencti-front/src/private/components/common/files/workbench/WorkbenchFileLine.jsx:321]: This file could maybe be refacto into typescript ?

**Review by SarahBocognano — COMMENTED** [review]
Tested ✅ 
The only question I have is that when I validate the workbench, the import is pending undefinetly...

File: opencti-platform/opencti-front/src/private/components/common/files/workbench/WorkbenchFileContent.jsx
Comment (helene-nguyen) [reply opencti-platform/opencti-front/src/private/components/common/files/workbench/WorkbenchFileContent.jsx:375]: As discussed with @Kedae, we used Ramda here to simplify the code

File: opencti-platform/opencti-front/src/private/components/common/files/workbench/WorkbenchFileLine.jsx
Comment (SouadHadjiat) [reply opencti-platform/opencti-front/src/private/components/common/files/workbench/WorkbenchFileLine.jsx:321]: Workbench will be completely refactored in the future, so not a priority to refactor into typescript this time.

File: opencti-platform/opencti-front/src/private/components/common/files/workbench/WorkbenchFileContent.jsx
Comment (helene-nguyen) [reply opencti-platform/opencti-front/src/private/components/common/files/workbench/WorkbenchFileContent.jsx:375]: > Tested ✅ The only question I have is that when I validate the workbench, the import is pending undefinetly...

Have you checked that the ImportDocument connector is active ?

Comment (SouadHadjiat) [reply opencti-platform/opencti-front/src/private/components/common/files/workbench/WorkbenchFileContent.jsx:375]: For workbench validation, I think it's `import-file-stix` connector that needs to be connected @SarahBocognano

Comment (SarahBocognano) [reply opencti-platform/opencti-front/src/private/components/common/files/workbench/WorkbenchFileContent.jsx:375]: It was the import-file-stix

Comment (SarahBocognano) [reply opencti-platform/opencti-front/src/private/components/common/files/workbench/WorkbenchFileContent.jsx:375]: It's working fine now :)

**Review by SarahBocognano — APPROVED** [review]

**Pull request merged**
